src/Microsoft.ComponentDetection.Common/FileWritingService.cs

```diff
@@ -26,10 +28,13 @@ public void AppendToFile(string relativeFilePath, string text)
     {
         relativeFilePath = this.ResolveFilePath(relativeFilePath);
 
-        lock (this.lockObject)
+        if (!this.bufferedStreams.TryGetValue(relativeFilePath, out var streamWriter))
```

Comment:
I know there are `async` overloads for writing to streams. I assume it would be a major refactoring to use them here? Can you create an issue for us to follow up with?

Reply:
It would be a huge refactor as every log statement would have to become async. We should just switch to a proper logging library like Serilog or the extension logging library instead, like we discussed offline.

Reply:
#436